lib/content-services/src/lib/search/components/search-date-range-advanced-tabbed/search-date-range-advanced/search-date-range-advanced.component.ts

```diff
@@ -0,0 +1,168 @@
+        inLastValueType: InLastDateType.DAYS,
+        inLastValue: undefined,
+        betweenStartDate: undefined,
+        betweenEndDate: undefined
+    });
+    betweenStartDateFormControl = this.form.controls.betweenStartDate;
+    betweenEndDateFormControl = this.form.controls.betweenEndDate;
+    convertedMaxDate: Date;
+    private destroy$ = new Subject<void>();
+
+    readonly DateRangeType = DateRangeType;
+    readonly InLastDateType = InLastDateType;
+
+    constructor(private formBuilder: FormBuilder,
+                private userPreferencesService: UserPreferencesService,
+                private dateAdapter: DateAdapter<DateFnsAdapter>,
+                @Inject(MAT_DATE_FORMATS) private dateFormatConfig: MatDateFormats) {}
+
+    readonly endDateValidator = (formControl: UntypedFormControl): ({ [key: string]: boolean } | null) => {
+        if (isBefore(formControl.value, this.betweenStartDateFormControl.value) || isAfter(formControl.value, this.convertedMaxDate)) {
+            return {
+                invalidDate: true
+            };
+        }
+        return {};
+    };
+
+    ngOnInit(): void {
+        this.dateFormatConfig.display.dateInput = this.dateFormat;
+        this.convertedMaxDate = endOfDay(this.maxDate && this.maxDate !== 'today' ?
+            parse(this.maxDate, this.dateFormat, new Date()) : new Date());
+        this.userPreferencesService
+            .select(UserPreferenceValues.Locale)
+            .pipe(takeUntil(this.destroy$))
+            .subscribe(locale => this.dateAdapter.setLocale(DateFnsUtils.getLocaleFromString(locale)));
+        this.form.controls.dateRangeType.valueChanges.pipe(takeUntil(this.destroy$))
+            .subscribe((dateRangeType) => this.updateValidators(dateRangeType));
+        this.form.valueChanges.pipe(takeUntil(this.destroy$))
+            .subscribe(() => this.onChange());
+    }
+
+    ngOnDestroy() {
+        this.destroy$.next();
+        this.destroy$.complete();
+    }
+
+    private updateValidators(dateRangeType: DateRangeType) {
+        switch(dateRangeType) {
+            case DateRangeType.BETWEEN:
+                this.betweenStartDateFormControl.setValidators(Validators.required);
+                this.betweenEndDateFormControl.setValidators([Validators.required, this.endDateValidator]);
+                this.form.controls.inLastValue.clearValidators();
+                break;
+            case DateRangeType.IN_LAST:
+                this.form.controls.inLastValue.setValidators(Validators.required);
+                this.betweenStartDateFormControl.clearValidators();
+                this.betweenEndDateFormControl.clearValidators();
+                break;
+            case DateRangeType.ANY:
+                this.form.controls.inLastValue.clearValidators();
+                this.betweenStartDateFormControl.clearValidators();
+                this.betweenEndDateFormControl.clearValidators();
+                break;
+            default:
```

Comment:
Default is the same like ANY so I would suggest to remove 'ANY' case from switch and leave only default. It will be fine because dateRangeType is an enum so it can't have more types so we can treat it as just default

Reply:
Removed ANY case from switch